tests/suites/test_suite_mpi.function

```diff
@@ -785,24 +789,12 @@ void mpi_core_cond_assign( data_t * input_X,
     TEST_EQUAL( limbs_X, limbs_Y );
     TEST_ASSERT( copy_limbs <= limbs );
 
-    ASSERT_ALLOC( X, limbs );
-    ASSERT_ALLOC( Y, limbs );
-
-    TEST_ASSERT( mbedtls_mpi_core_read_be( X, limbs, input_X->x, input_X->len )
-                 == 0 );
-
-    TEST_ASSERT( mbedtls_mpi_core_read_be( Y, limbs, input_Y->x, input_Y->len )
-                 == 0 );
-
     /* condition is false */
     TEST_CF_SECRET( X, bytes );
     TEST_CF_SECRET( Y, bytes );
 
     mbedtls_mpi_core_cond_assign( X, Y, copy_limbs, 0 );
 
-    TEST_CF_PUBLIC( X, bytes );
-    TEST_CF_PUBLIC( Y, bytes );
-
     TEST_ASSERT( memcmp( X, Y, bytes ) != 0 );
```

Comment:
You can remove `TEST_CF_PUBLIC` before `TEST_EQUAL`, `TEST_LE_U` and `TEST_LE_S`, and we could do the same with `ASSERT_COMPARE` but haven't done so yet (with this one, I'm worried that sometimes we do want to keep the buffer contents private), but you need to make the buffer public before calling `memcmp`.

Reply:
I have tested and `TEST_CF_PUBLIC` is needed before `ASSERT_COMPARE`. 
The `ASSERT_COMPARE` is also calling `memcmp` without `TEST_CF_PUBLIC`.